### PR TITLE
Add env substitution to `pathorcontent`

### PR DESCRIPTION
**Pull request by saswatamcode**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes
This PR adds environment variable substitution to `pathorcontent` flag by replacing `pkg/extflag/pathorcontent.go` with `github.com/efficientgo/tools/extkingpin`. This was added as a new module [here](https://github.com/efficientgo/tools/pull/10). Addresses #4293.

## Verification
Tested locally.

### CODE REVIEW

**Comment by saswatamcode** [comment]
Test failure seems to be unrelated (timeout).

**Review by bwplotka — APPROVED** [review]
This LGTM, but let's wait for others 👀 too.

@thanos-io/thanos-maintainers this moves pathOrContent to separate reusable go module BUT it also adds environment variable substitution which makes our flags with config accept environment variables within YAML config using `$()`

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ We use _breaking :warning:_ to mark changes that are not backward compatible (re
 ### Added
 
 - [#4299](https://github.com/thanos-io/thanos/pull/4299) Tracing: Add tracing to exemplar APIs.
+- [#4327](https://github.com/thanos-io/thanos/pull/4327) Add env substitution to PathOrContent flag.
```
Comment (kakkoyun) [inline CHANGELOG.md:18]: The changelog for the users' perspective, so it'd be nice to tell user that they can use env variable for certain flag argument.s

**Review by kakkoyun — APPROVED** [review]
LGTM 🚀 

Only one nit about the changelog.

**Review by bwplotka — APPROVED** [review]

**Pull request merged**